### PR TITLE
Add unselected option to select widget when no default is set

### LINKED ISSUE #562: Select widget displays first option by default, but behaves as if empty
**- Do you want to request a *feature* or report a *bug*?**
bug

**- What is the current behavior?**
When you start a new item in a collection, select widgets (without a set default - haven't tried with a default yet) will display the first option, as if they are selected. However, when you try to save, the field behaves as if it's blank: saving is blocked, and the label turns red.

**- If the current behavior is a bug, please provide the steps to reproduce.**
Make a new Kitchen Sink item in the CMS demo site: https://cms-demo.netlify.com/#/collections/kitchenSink/entries/new

Note that the Select field appears to be set to the first choice, "a". Click save, watch it fail, and see the red label for the select entry:
filledselect

Select any option (inlcuding "a"), and it will save properly.

**- What is the expected behavior?**
Normally, I would expect the dropdowns to be blank in their initial state. If not, they should actually be set to the option they display.

**- Please mention your node.js, and operating system version.**
Current Chrome on Windows 10, CMS 0.4.6

### PR DESCRIPTION
**Pull request by Benaiah**
**- Summary**

Add a "not selected" option to the select widget when there is no default set. Closes #562.

**- Test plan**

Tested manually.

**- Description for the changelog**

Add unselected option to select widget when no default is set.

**- A picture of a cute animal (not mandatory but encouraged)**

image

### CODE REVIEW

File: src/components/Widgets/SelectControl.js [hunk 1]
```diff
@@ -15,12 +15,15 @@ export default class SelectControl extends React.Component {
...
     }
 
-    const options = fieldOptions.map((option) => {
-      if (typeof option === 'string') {
-        return { label: option, value: option };
-      }
-      return option;
-    });
+    const options = [
+      ...(field.get('default', false) ? [] : [{ label: '<not set>', value: '' }]),
```
Comment (tech4him1) [inline src/components/Widgets/SelectControl.js]: I don't think the `<not set>` field should be a regular option -- I don't think it should be selectable. ~~After an option has been selected, the `<not set>` field should be hidden.~~

Comment (tech4him1) [reply src/components/Widgets/SelectControl.js]: I would change `{ label: '<not set>', value: '' }` to `{ label: '<not set>', disabled: true }`.

Comment (Benaiah) [reply src/components/Widgets/SelectControl.js]: @tech4him1 good point, I'll make the change.

Comment (tech4him1) [reply src/components/Widgets/SelectControl.js]: @Benaiah actually, now that you say that I'm not sure it does. If I use this PR like it is right now, it won't let me save if I select `<not set>`.

Comment (Benaiah) [reply src/components/Widgets/SelectControl.js]: @tech4him1 "not sure it does" :point_left: not sure it does what?

Comment (tech4him1) [reply src/components/Widgets/SelectControl.js]: @Benaiah There is no way, AFAIK, to save a select with an empty value unless `required: false`. Is that what you meant?

Comment (Benaiah) [reply src/components/Widgets/SelectControl.js]: @tech4him1 I meant that if you want to save an empty string, you can always add an option like `{ label: "Empty", value: "" }` to the config. `<not set>` shouldn't be selectable, as you suggested.

Comment (tech4him1) [reply src/components/Widgets/SelectControl.js]: Agreed.

**Comment by Benaiah** [comment]
@tech4him1, @erquhart this is finished and r2r.

**Pull request merged**